### PR TITLE
Add reduce_max_grad composite rule

### PR DESCRIPTION
**Pull request by wangxn12138**
### PR types

New features
### PR changes

Others
### Describe

add reduce_max_grad composite rule.
becuse original op test pass fp16, composite op test pass fp16 too.

### CODE REVIEW

**Comment by cyber-pioneer** [comment]
pr name typo

**Comment by wangxn12138** [comment]
> pr name typo

done

File: python/paddle/fluid/tests/unittests/test_reduce_op.py [hunk 1]
```diff
@@ -243,6 +244,10 @@ def setUp(self):
     def test_check_output(self):
         self.check_output(check_eager=True)
 
+    def test_check_grad(self):
+        # only composite op support gradient check of reduce_max
+        self.check_grad(['X'], 'Out', check_eager=True, only_check_prim=True)
+
     def test_raise_error(self):
```
Comment (xiaoguoguo626807) [inline python/paddle/fluid/tests/unittests/test_reduce_op.py:258]: if original test pass fp16, please describe it in pr describes

File: python/paddle/fluid/tests/unittests/test_reduce_op.py [hunk 2]
```diff
@@ -271,6 +277,10 @@ def setUp(self):
     def test_check_output(self):
         self.check_output(check_eager=True)
 
+    def test_check_grad(self):
```
Comment (xiaoguoguo626807) [inline python/paddle/fluid/tests/unittests/test_reduce_op.py:289]: why original op test don't support grad check

File: python/paddle/fluid/tests/unittests/test_reduce_op.py [hunk 3]
```diff
@@ -261,6 +266,7 @@ class TestMaxOp_ZeroDim(OpTest):
 
     def setUp(self):
         self.op_type = "reduce_max"
+        self.prim_op_type = "prim"
         self.python_api = paddle.max
         self.inputs = {'X': np.random.random([]).astype("float64")}
         self.attrs = {'dim': []}
```
Comment (xiaoguoguo626807) [inline python/paddle/fluid/tests/unittests/test_reduce_op.py:281]: it's better add new op test to check keep_dim = True and FP32

File: python/paddle/fluid/tests/unittests/test_reduce_op.py [hunk 4]
```diff
@@ -233,6 +233,7 @@ class TestMaxOp(OpTest):
 
     def setUp(self):
         self.op_type = "reduce_max"
+        self.prim_op_type = "prim"
         self.python_api = paddle.max
         self.inputs = {'X': np.random.random((5, 6, 10)).astype("float64")}
         self.attrs = {'dim': [-1]}
```
Comment (xiaoguoguo626807) [inline python/paddle/fluid/tests/unittests/test_reduce_op.py:240]: it's better to check keep_dim = True

File: python/paddle/fluid/tests/unittests/test_reduce_op.py (hunk 2 above)
Comment (xiaoguoguo626807) [reply python/paddle/fluid/tests/unittests/test_reduce_op.py:289]: @Charles-hit  if original op has no grad test how to compare with composite_grad

Comment (wangxn12138) [reply python/paddle/fluid/tests/unittests/test_reduce_op.py:289]: > why original op test don't support grad check

装饰器注明"reduce_max is discontinuous non-derivable function, its gradient check is not supported by unittest framework."

File: python/paddle/fluid/tests/unittests/test_reduce_op.py (hunk 1 above)
Comment (wangxn12138) [reply python/paddle/fluid/tests/unittests/test_reduce_op.py:258]: > if original test pass fp16, please describe it in pr describes

done

File: python/paddle/fluid/tests/unittests/test_reduce_op.py (hunk 3 above)
Comment (wangxn12138) [reply python/paddle/fluid/tests/unittests/test_reduce_op.py:281]: > it's better add new op test to check keep_dim = True and FP32

done

File: python/paddle/fluid/tests/unittests/test_reduce_op.py (hunk 4 above)
Comment (wangxn12138) [reply python/paddle/fluid/tests/unittests/test_reduce_op.py:239]: done

Comment (wangxn12138) [reply python/paddle/fluid/tests/unittests/test_reduce_op.py:239]: > Contributor

done

Comment (wangxn12138) [reply python/paddle/fluid/tests/unittests/test_reduce_op.py:240]: > it's better to check keep_dim = True

done

File: paddle/fluid/prim/api/api.yaml [hunk 5]
```diff
@@ -37,6 +37,8 @@
 - pad
 - cumsum
 - put_along_axis
+- equal
+- where
```
Comment (xiaoguoguo626807) [inline paddle/fluid/prim/api/api.yaml]: two where， delete one

Comment (wangxn12138) [reply paddle/fluid/prim/api/api.yaml]: > two where， delete one

done

**Review by Charles-hit — APPROVED** [review]

**Review by xiaoguoguo626807 — APPROVED** [review]

**Pull request merged**